include/seqan3/alphabet/detail/alphabet_proxy.hpp

```diff
@@ -95,6 +93,37 @@ class alphabet_proxy : public alphabet_type
 
 #if 1// this is the one that works for most things, but not all
 
+namespace seqan3::detail
+{
+
+//!\brief Auxiliary transformation trait that is void for types by default.
+//!\ingroup alphabet
+template <typename t>
+struct alphabet_char_type_or_void
```

Comment:
We don't have the `underlying_char` without `_t` anymore, that's why the change was necessary. The previous design suggested that it was a customisation point, because you could specialise it (which it was before). Now that it's not any longer, I wanted to remove the possibility (the template alias cannot be specialised by definition).